File: src/types/treatment.ts
export interface TreatmentPlan {
  id: number;
  hn: string;
  regimen: string;
  phase: 'intensive' | 'continuation';
  phase_start: string;
  phase_end_expected: string | null;
  drugs: string; // JSON array string
  duration_months: number;
  is_current: boolean;
  notes: string | null;
  created_at: string;
}

export interface TreatmentPlanUpdate {
  hn: string;
  new_phase: string;
  phase_start: string;
  regimen: string;
  drugs: string;
  duration_months: number;
  notes: string | null;
}

export interface Followup {
  id: number;
  hn: string;
  followup_date: string;
  month_number: number | null;
  weight_kg: number | null;
  sputum_result: 'negative' | 'positive' | 'not_done' | null;
  xray_result: 'improved' | 'stable' | 'worse' | 'not_done' | null;
  side_effects: string | null; // JSON
  adherence: 'good' | 'fair' | 'poor' | null;
  dispensed_drugs: string | null; // JSON
  notes: string | null;
  created_by: string | null;
  created_at: string;
}

export interface FollowupInput {
  hn: string;
  followup_date: string;
  month_number: number | null;
  weight_kg: number | null;
  sputum_result: string | null;
  xray_result: string | null;
  side_effects: string[] | null;
  adherence: string | null;
  notes: string | null;
  created_by: string | null;
}

export interface Outcome {
  id: number;
  hn: string;
  outcome: string;
  outcome_date: string;
  treatment_end: string | null;
  notes: string | null;
  created_by: string | null;
  created_at: string;
}

export interface OutcomeInput {
  hn: string;
  outcome: string;
  outcome_date: string;
  treatment_end: string | null;
  notes: string | null;
  created_by: string | null;
}

File: crates/tb-models/src/treatment.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone, sqlx::FromRow)]
pub struct TreatmentPlan {
  pub id: i64,
  pub hn: String,
  pub regimen: String,
  pub phase: String,
  pub phase_start: String,
  pub phase_end_expected: Option<String>,
  pub drugs: String,
  pub duration_months: i64,
  pub is_current: bool,
  pub notes: Option<String>,
  pub created_at: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct TreatmentPlanUpdate {
  pub hn: String,
  pub new_phase: String,
  pub phase_start: String,
  pub regimen: String,
  pub drugs: String,
  pub duration_months: i64,
  pub notes: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone, sqlx::FromRow)]
pub struct Followup {
  pub id: i64,
  pub hn: String,
  pub followup_date: String,
  pub month_number: Option<i64>,
  pub weight_kg: Option<f64>,
  pub sputum_result: Option<String>,
  pub xray_result: Option<String>,
  pub side_effects: Option<String>,
  pub adherence: Option<String>,
  pub dispensed_drugs: Option<String>,
  pub notes: Option<String>,
  pub created_by: Option<String>,
  pub created_at: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct FollowupInput {
  pub hn: String,
  pub followup_date: String,
  pub month_number: Option<i64>,
  pub weight_kg: Option<f64>,
  pub sputum_result: Option<String>,
  pub xray_result: Option<String>,
  pub side_effects: Option<Vec<String>>,
  pub adherence: Option<String>,
  pub notes: Option<String>,
  pub created_by: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone, sqlx::FromRow)]
pub struct Outcome {
  pub id: i64,
  pub hn: String,
  pub outcome: String,
  pub outcome_date: String,
  pub treatment_end: Option<String>,
  pub notes: Option<String>,
  pub created_by: Option<String>,
  pub created_at: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct OutcomeInput {
  pub hn: String,
  pub outcome: String,
  pub outcome_date: String,
  pub treatment_end: Option<String>,
  pub notes: Option<String>,
  pub created_by: Option<String>,
}
